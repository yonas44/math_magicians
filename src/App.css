.App-wrapper {
  background: url(./assets/jeswin-thomas-hecib2an4T4-unsplash.jpg);
  background-position: center;
  background-size: cover;
  width: 100vw;
  height: 100vh;
}

.App {
  backdrop-filter: blur(8px);
  display: flex;
  flex-direction: column;
  width: 100vw;
  height: 100vh;
}
