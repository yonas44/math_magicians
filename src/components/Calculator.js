import React, { useEffect, useState } from 'react';
import Button from './button/buttonComponent';
import buttonKeys from './data';
import './styles/calculator.css';
import calculate from '../logic/calculate';

const Calculator = () => {
  const [buttons, setButtons] = useState([]);
  const [results, setResults] = useState({});

  useEffect(() => {
    setButtons(buttonKeys);
    return () => {
      setButtons([]);
    };
  }, []);

  const handleChange = (input) => {
    const values = calculate(results, input.target.textContent);
    const newResult = { ...results };
    newResult.total = values.total;
    newResult.next = values.next;
    newResult.operation = values.operation || null;
    setResults(newResult);
  };

  return (
    <div className="calculator-container">
      <p id="slogan">Let&apos;s do some math!</p>
      <div className="calculator">
        <div className="result-screen">
          <p data-testid="total" id="total">
            {results.total}
          </p>
          <p id="op">{results.operation}</p>
          <p id="next">{results.next || 0}</p>
        </div>
        <div className="button-section">
          {buttons.map((buttonkey) => (
            <Button
              key={buttonkey.value}
              value={buttonkey.value}
              role={buttonkey.type}
              handleChange={handleChange}
            />
          ))}
        </div>
      </div>
    </div>
  );
};

export default Calculator;
