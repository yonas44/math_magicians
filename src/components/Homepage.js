import React from 'react';
import './styles/Homepage.css';

const Homepage = () => (
  <div className="container">
    <h2 className="title">Welcome to our page!</h2>
    <p className="description">
      Up am intention on dependent questions oh elsewhere september. No betrayed
      pleasure possible jointure we in throwing. And can event rapid any shall
      woman green. Hope they dear who its bred. Smiling nothing affixed he
      carried it clothes calling he no. Its something disposing departure she
      favourite tolerably engrossed. Truth short folly court why she their
      balls. Excellence put unaffected reasonable mrs introduced conviction she.
      Nay particular delightful but unpleasant for uncommonly who. He went such
      dare good mr fact. The small own seven saved man age no offer. Suspicion
      did mrs nor furniture smallness. Scale whole downs often leave not eat. An
      expression reasonably cultivated indulgence mr he surrounded instrument.
      Gentleman eat and consisted are pronounce distrusts.
    </p>
  </div>
);

export default Homepage;
