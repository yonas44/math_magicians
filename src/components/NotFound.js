import React from 'react';
import './styles/NotFound.css';

const NotFound = () => (
  <div className="notfound-container">
    <h2 id="notfound-title">404</h2>
    <p>_Page not found_</p>
  </div>
);

export default NotFound;
