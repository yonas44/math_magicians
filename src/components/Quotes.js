import React, { useState, useEffect } from 'react';
import { MdChangeCircle } from 'react-icons/md';
import './styles/Quotes.css';

const Quotes = () => {
  const [quote, setQuote] = useState({ author: null, content: 'Your quote is being fetchedðŸ˜€' });

  const fetchQuotes = () => {
    fetch('https://quotable.io/random')
      .then((response) => response.json())
      .then((response) => setQuote((prevQuote) => {
        const newQuote = { ...prevQuote };
        newQuote.author = response.author;
        newQuote.content = response.content;
        return newQuote;
      }));
  };

  useEffect(() => {
    fetchQuotes();
  }, []);
  return (
    <>
      <div data-testid="quotes" className="quote-container">
        <button id="new-quote" type="button" onClick={() => fetchQuotes()}>
          <MdChangeCircle style={{ scale: '2' }} />
        </button>
        <p id="quote-content">
          &apos;
          {quote.content}
          &apos;
        </p>
        <p id="quote-author">{quote.author}</p>
      </div>
    </>
  );
};

export default Quotes;
