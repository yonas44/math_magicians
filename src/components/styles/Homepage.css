* {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
}

.container {
  width: 80%;
  margin: 50px auto;
}

.title {
  margin: 20px 0;
  font-weight: bolder;
}

@media screen and (min-width: 768px) {
  .container {
    margin: 200px auto;
  }
}
