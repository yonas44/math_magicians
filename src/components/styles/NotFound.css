* {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
}

.notfound-container {
  text-align: center;
}

#notfound-title {
  color: red;
  font-size: 62px;
}
