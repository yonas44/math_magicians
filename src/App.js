import { Routes, Route } from 'react-router-dom';
import './App.css';
import Calculator from './components/Calculator';
import Homepage from './components/Homepage';
import Navbar from './components/Navbar';
import NotFound from './components/NotFound';
import Quotes from './components/Quotes';

function App() {
  return (
    <div className="App-wrapper">
      <div className="App">
        <Routes>
          <Route element={<Navbar />} path="/">
            <Route element={<Calculator />} path="/calculator" />
            <Route element={<Homepage />} exact path="/" />
            <Route element={<Quotes />} exact path="/quotes" />
          </Route>
          <Route element={<NotFound />} exact path="*" />
        </Routes>
      </div>
    </div>
  );
}

export default App;
