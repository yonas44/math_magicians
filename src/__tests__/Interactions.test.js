import { BrowserRouter, Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';
import { render, screen, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Calculator from '../components/Calculator';
import Homepage from '../components/Homepage';
import App from '../App';
import NotFound from '../components/NotFound';
import Quotes from '../components/Quotes';

describe('HomePage component is rendererd', () => {
  test('Component is rendererd', () => {
    render(<Homepage />);
    const container = screen.getByText(/Welcome to our page!/);
    expect(container).toBeInTheDocument();
  });

  test('matches snapshot', () => {
    const tree = renderer.create(<Homepage />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Calculator component is rendererd', () => {
  test('Component functions', () => {
    // Arrange
    render(<Calculator />);

    // Act
    fireEvent.click(screen.getByText('1'));
    fireEvent.click(screen.getByText('2'));
    fireEvent.click(screen.getByText('+'));
    fireEvent.click(screen.getByText('3'));
    fireEvent.click(screen.getByText('='));
    const result = screen.getByTestId('total');

    // Assert
    expect(result).toHaveTextContent('15');
  });

  test('matches snapshot', () => {
    const tree = renderer.create(<Calculator />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

// Routing test

describe('Navbar renders', () => {
  test('should redirect and update history', () => {
    const history = createMemoryHistory();
    render(
      <Router location={history.location} navigator={history}>
        <App />
      </Router>,
    );

    fireEvent.click(screen.getByText(/Calculator/));
    expect(history.location.pathname).toEqual('/calculator');
  });

  test('should redirect and update dom', () => {
    render(
      <BrowserRouter>
        <App />
      </BrowserRouter>,
    );

    fireEvent.click(screen.getByText(/Calculator/));
    expect(screen.getByText(/Let's do some math!/)).toBeInTheDocument();
  });
});

describe('Not-found page is rendered', () => {
  test('Component is rendered', () => {
    render(<NotFound />);
    const container = screen.getByText(/404/);
    expect(container).toBeInTheDocument();
  });

  test('matches snapshot', () => {
    const tree = renderer.create(<NotFound />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Quotes page is rendered', () => {
  test('Component is rendered', () => {
    render(<Quotes />);
    const container = screen.getByTestId('quotes');
    expect(container).toBeInTheDocument();
  });

  test('matches snapshot', () => {
    const tree = renderer.create(<Quotes />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
