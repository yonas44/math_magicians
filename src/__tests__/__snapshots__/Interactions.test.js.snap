// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator component is rendererd matches snapshot 1`] = `
<div
  className="calculator-container"
>
  <p
    id="slogan"
  >
    Let's do some math!
  </p>
  <div
    className="calculator"
  >
    <div
      className="result-screen"
    >
      <p
        data-testid="total"
        id="total"
      />
      <p
        id="op"
      />
      <p
        id="next"
      >
        0
      </p>
    </div>
    <div
      className="button-section"
    />
  </div>
</div>
`;

exports[`HomePage component is rendererd matches snapshot 1`] = `
<div
  className="container"
>
  <h2
    className="title"
  >
    Welcome to our page!
  </h2>
  <p
    className="description"
  >
    Up am intention on dependent questions oh elsewhere september. No betrayed pleasure possible jointure we in throwing. And can event rapid any shall woman green. Hope they dear who its bred. Smiling nothing affixed he carried it clothes calling he no. Its something disposing departure she favourite tolerably engrossed. Truth short folly court why she their balls. Excellence put unaffected reasonable mrs introduced conviction she. Nay particular delightful but unpleasant for uncommonly who. He went such dare good mr fact. The small own seven saved man age no offer. Suspicion did mrs nor furniture smallness. Scale whole downs often leave not eat. An expression reasonably cultivated indulgence mr he surrounded instrument. Gentleman eat and consisted are pronounce distrusts.
  </p>
</div>
`;

exports[`Not-found page is rendered matches snapshot 1`] = `
<div
  className="notfound-container"
>
  <h2
    id="notfound-title"
  >
    404
  </h2>
  <p>
    _Page not found_
  </p>
</div>
`;

exports[`Quotes page is rendered matches snapshot 1`] = `
<div
  className="quote-container"
  data-testid="quotes"
>
  <button
    id="new-quote"
    onClick={[Function]}
    type="button"
  >
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
          "scale": "2",
        }
      }
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
      <path
        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm.06 17v-2.01H12c-1.28 0-2.56-.49-3.54-1.46a5.006 5.006 0 01-.64-6.29l1.1 1.1c-.71 1.33-.53 3.01.59 4.13.7.7 1.62 1.03 2.54 1.01v-2.14l2.83 2.83L12.06 19zm4.11-4.24l-1.1-1.1c.71-1.33.53-3.01-.59-4.13A3.482 3.482 0 0012 8.5h-.06v2.15L9.11 7.83 11.94 5v2.02c1.3-.02 2.61.45 3.6 1.45 1.7 1.7 1.91 4.35.63 6.29z"
      />
    </svg>
  </button>
  <p
    id="quote-content"
  >
    '
    Your quote is being fetchedðŸ˜€
    '
  </p>
  <p
    id="quote-author"
  />
</div>
`;
